### DISCUSSION TITLE
Argocd cannot communicate with github

### DISCUSSION THREAD

**Discussion by calvinShan** [argoproj/argo-cd#7283] [Q&A]
Hello, 
After installing argocd on AKS Azure, we cannot creat the appliction, because it cannot comunicate with github.

error: FATA[0040] rpc error: code = InvalidArgument desc = application spec is invalid: InvalidSpecError: repository not accessible: rpc error: code = Unknown desc = Get "http://xxxl/xxx/configurationkubernetes/info/refs?service=git-upload-pack": context deadline exceeded (Client.Timeout exceeded while awaiting headers)

How could we resolve this porblem, maybe we need to configure environments variables during installing?

how to configure these variables by changing the install.yaml and using kubectl?

Thanks

**Comment by wanghong230** [comment]
Could you share more detail? I feel it could be the connectivity issue from the cluster to your Git repo? Could you run a debug pod in the namespace to see whether you can reach your git server?  Do you use GitHub Enterprise?

**Reply by calvinShan** [reply]
hello @wanghong230 
Yes, when I exec the pod, i can clone the projet Gitlab to local environment.
And we could create application  argocd by using ip adress instead of hostname of gitlab.

So the problem is about the DNS mapping. However, we could sucessfully clone the project inside the pod of argocd, so maybe we need to configure DNS for argocd service?

Inside pod argo : argocd-server pod
clone hostname/projet ok
clone IP/project OK

From argocd interface (chrome)
clone hostname/projet **KO**
clone IP/project OK

Thanks

**Reply by Mcduller** [reply]
@calvinShan hello, i met the same question, did you solve it?
